gems/quilt_rails/README.md

```diff
@@ -0,0 +1,82 @@
+# quilt_rails
+
+A turn-key solution for integrating server-rendered react into your Rails app using Quilt libraries.
+
+This document focuses on Rails integration. For details of `@shopify/react-server`'s configuration and usage, see the [react-server documentation](/packages/react-server/README.md).
+
+## Quick Start
+
+Create a Rails project using `dev init` then:
+
+### Install Dependencies
+
+```sh
+# Add Ruby/Node dependencies
+bundle add sewing_kit quilt_rails
+yarn add @shopify/sewing-kit @shopify/react-server
+
+# Optional - add Polaris and quilt libraries
+yarn add @shopify/polaris @shopify/react-self-serializers react react-dom
+
+yarn
+dev up
+```
+
+### Add JavaScript
+
+sewing_kit looks for JavaScript in `app/ui/index.js`. The code in `index.js` (and any imported JS/CSS) will be built into a `main` bundle.
+
+### Setup your react controller and routes
+
+Create a `ReactController` to handle react requests.
+
+```ruby
+class ReactController < ApplicationController
```

Comment:
I intentionally left it this way rather than as an engine so that consuming apps can setup their own `before_action`s and whatnot, since part of the goal of this is to let Rails devs continue to handle authentication/redirects and whatever other concerns they might want to at the request level if they want to.

I guess another alternative would be to provide both options.

Reply:
I think this is very good.